rest/data/project_settings.go

```diff
@@ -410,12 +410,60 @@ func SaveProjectSettingsForSection(ctx context.Context, projectId string, change
 		modified, err = updateAliasesForSection(projectId, changes.Aliases, before.Aliases, section)
 		catcher.Add(err)
 	case model.ProjectPageNotificationsSection:
-		if err = SaveSubscriptions(projectId, changes.Subscriptions, true); err != nil {
+		subscriptionChanges := []restModel.APISubscription{}
+		for _, subscription := range changes.Subscriptions {
+			webhook := subscription.Subscriber.WebhookSubscriber
+			if webhook == nil {
+				subscriptionChanges = append(subscriptionChanges, subscription)
+				continue
+			}
+			// If this subscription is a webhook, we should redact the webhook secret and Authorization header.
+			// This means when the user changes other values, we should not update the secret or Authorization header if
+			// they are still set to the redacted value and instead we should use the previous value(s).
+			var previousSubscription *event.WebhookSubscriber
+			for _, beforeSubscription := range before.Subscriptions {
+				if beforeSubscription.ID == utility.FromStringPtr(subscription.ID) {
+					var ok bool
+					previousSubscription, ok = beforeSubscription.Subscriber.Target.(*event.WebhookSubscriber)
+					if !ok {
+						return nil, errors.Errorf("could not find subscription with ID '%s'", utility.FromStringPtr(subscription.ID))
+					}
+					break
+				}
+			}
+			if previousSubscription == nil {
+				// If there are no previous subscriptions, we should just add the new subscription.
+				subscriptionChanges = append(subscriptionChanges, subscription)
+				continue
+			}
+			if webhook.Secret != nil && *webhook.Secret == evergreen.RedactedWebhookSecretsValue {
```

Comment:
nit: looks like this could be simplified down to `utility.FromStringPtr(webhook.Secret) == evergreen.RedactedWebhookSecretsValue`.